modules/core/src/main/scala/iota/internal/ProductMacros.scala

```diff
@@ -17,6 +17,51 @@ import scalaz.NonEmptyList
 import scalaz.std.list._
 //#-scalaz
 
+final class ProductSeq(p: Product) extends OptimisedIndexedSeq[Any] {
```

Comment:
Should this be removed too?

Reply:
no, you'll want to use it when you port over my Iso things. I'd do it in this PR but it will only work in the scalaz code path.

Reply:
Got it. I'll merge this as-is.